src/sagemaker_training/process.py

```diff
@@ -111,26 +143,35 @@ async def run_async(cmd, processes_per_host, env, cwd, stderr, **kwargs):
         asyncio.subprocess.Process: The asyncio process for the given command.
 
     Raises:
-        error_class: If there is an exception raised when creating the process.
+        ExecuteUserScriptError: If there is an exception raised when creating the process.
     """
     cmd = " ".join(cmd)
     proc = await asyncio.create_subprocess_shell(
         cmd, env=env, cwd=cwd, stdout=PIPE, stderr=stderr, **kwargs
     )
 
     output = await asyncio.gather(
-        watch(proc.stdout, processes_per_host), watch(proc.stderr, processes_per_host)
+        watch(proc.stdout, processes_per_host, error_classes=error_classes),
+        watch(proc.stderr, processes_per_host, error_classes=error_classes),
     )
     return_code = proc.returncode
     return return_code, output, proc
 
 
-def create(cmd, error_class, processes_per_host, cwd=None, env=None, capture_error=False, **kwargs):
+def create(
+    cmd,
+    error_classes,
+    processes_per_host,
+    cwd=None,
+    env=None,
+    capture_error=False,
+    **kwargs,
+):
     """Spawn a process with asyncio for the given command.
 
     Args:
         cmd (list): The command to be run.
-        error_class (cls): The class to use when raising an exception.
+        error_classes (list): List of exception classes to watch and raise.
```

Comment:
Looks like this is an optional list argument? When modelparallel is not enabled (in `mpi.py`, this is still a class and you are wrapping it into list on the fly in the watch function.

Reply:
Yes, it's for backward compatibility. I don't want to break any existing code, so made it optional list, also passing a single class would also work.